### PR TITLE
#24582: Remove compare product option is not working for out of stock products in magento2.3.2

### PR DESCRIPTION
**Pull request by sudheers-kensium**
#24582: Remove compare product option is not working for out of stock products in magento2.3.2 - fixed

### Description (*)

#24582: Remove compare product option is not working for out of stock products in magento2.3.2

### Fixed Issues (if relevant)

1. magento/magento2#24582:  Remove compare product option is not working for out of stock products in magento2.3.2

### Manual testing scenarios (*)

1. Create simple product in admin
2. Go to frontend and add product to compare list
3. Change product stock status from In Stock to Out Of Stock and save
4. Go to frontend compare list and remove out of stock product from compare list 
5. Product should remove from compare list and show message You removed product from the comparison list.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Catalog/Controller/Product/Compare/Remove.php [hunk 1]
```diff
@@ -31,7 +31,7 @@ public function execute()
                 $product = null;
             }
 
-            if ($product && $product->isSalable()) {
+            if ($product) {
```
Comment (kalpmehta) [inline app/code/Magento/Catalog/Controller/Product/Compare/Remove.php:34]: I am not sure if we should remove `$product->isSalable()`. In what case will someone compare the non-saleable product and is it correct to allow such comparisons?

Comment (kalpmehta) [reply app/code/Magento/Catalog/Controller/Product/Compare/Remove.php:34]: I see it is to remove the compared products, which might be the correct behavior even though the product is OOS. I will wait for someone else to review as well.

**Comment by kalpmehta** [comment]
As per the linked issue, this is not reproduce-able. I am closing this PR, but please feel free to re-open if you think otherwise.

**Comment by sudheers-kensium** [comment]
@kalpmehta Issue is reproducible on 2.3-develop instance and customer can able remove out of stock product from compare list

**Comment by sudheers-kensium** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @sudheers-kensium. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @sudheers-kensium, here is your [new Magento instance](https://pr-24583.instances.magento-community.engineering).
Admin access: [https://pr-24583.instances.magento-community.engineering/admin](https://pr-24583.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by sudheers-kensium** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @sudheers-kensium. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @sudheers-kensium, here is your [Magento instance](https://i-24583-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-24583-2-3-develop.instances.magento-community.engineering/admin](https://i-24583-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by lenaorobei** [comment]
Closing this PR since issue was fixed by internal team.

**Pull request closed without merging**